analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/LocksReleasedAllPathsBase.cs

```diff
@@ -179,12 +188,51 @@ private ProgramState RemoveLock(SymbolicContext context, ISymbol symbol)
         private static ProgramState AddLock(SymbolicContext context, IOperation operation) =>
             context.State.SetOperationConstraint(operation, context.SymbolicValueCounter, LockConstraint.Held);
 
-        private static ISymbol FirstArgumentSymbol(IInvocationOperationWrapper invocation) =>
-            IArgumentOperationWrapper.FromOperation(invocation.Arguments.First()).Value.TrackedSymbol();
+        private static ISymbol ArgumentSymbol(IInvocationOperationWrapper invocation, int argumentIndex, string parameterName) =>
```

Comment:
We should not need `parameterName` here, it's redundant. We can get the information from `invocation.TargetMethod.Parameters[argumentIndex].Name is var parameterName`.
```suggestion
        private static ISymbol ArgumentSymbol(IInvocationOperationWrapper invocation, int argumentIndex) =>
```

Reply:
While on it, we should rename `argumentIndex` to `parameterIndex`, because we want whatever argument that belongs to parameter with that index.